.github/workflows/resolve-build-deps.yaml

```
@@ -34,14 +27,76 @@ env:
   # https://reproducible-builds.org/specs/source-date-epoch/
   SOURCE_DATE_EPOCH: "1580601600"
 
-jobs:
+jobs: 
+  # measure-disk-usage.yml depends on this workflow being triggered and completed,
+  # so it can wait for the build to calculate dependency sizes.
+  # The 'on' setting ensures it runs, but this job cancels it if no dependency changes are detected.
+
+  check-dependency-changes:
+    name: Check dependency changes
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: write
+    outputs:
+      dependency_changed: ${{ steps.dependency-check.outputs.dependency_changed }}
+      builder_changed: ${{ steps.dependency-check.outputs.builder_changed }}
+    steps:
+      - name: Define diff commits
+        id: set_sha
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            PREV_SHA=${{ github.event.pull_request.base.sha }}
+            CURR_SHA=${{ github.event.pull_request.head.sha }}
+          else
+            PREV_SHA=${{ github.event.before }}
+            CURR_SHA=${{ github.sha }}
+          fi
+
+          echo "prev_sha=$PREV_SHA" >> $GITHUB_OUTPUT
+          echo "curr_sha=$CURR_SHA" >> $GITHUB_OUTPUT
+
+          echo "Current SHA: $CURR_SHA"
+          echo "Previous SHA: $PREV_SHA"
+
+      - name: Get changed files
+        id: changed-files
+        run: |
+          REPO="${{ github.repository }}"
+
+          CHANGED_FILES=$(gh api "repos/$REPO/compare/${{ steps.set_sha.outputs.prev_sha }}...${{ steps.set_sha.outputs.curr_sha }}" --paginate | \
+          jq -r '.files | map(.filename) | join(" ")')
+
+          echo "files_changed=$CHANGED_FILES" >> $GITHUB_OUTPUT
+          echo "files_changed=$CHANGED_FILES"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check for dependency changes
+        id: dependency-check
+        run: |
+          FILES_CHANGED="${{ steps.changed-files.outputs.files_changed }}"
+          DEPENDENCY_CHANGED=$(echo "$FILES_CHANGED" | grep -qE "(\.github/workflows/resolve-build-deps\.yml|\.builders/|agent_requirements\.in)" && echo "true" || echo "false") 
+          BUILDER_CHANGED=$(echo "$FILES_CHANGED" | grep -qE "^\.builders/" && echo "true" || echo "false")
+
+          echo "dependency_changed=$DEPENDENCY_CHANGED" >> $GITHUB_OUTPUT
+          echo "dependency_changed=$DEPENDENCY_CHANGED"
+
+          echo "builder_changed=$BUILDER_CHANGED" >> $GITHUB_OUTPUT
+          echo "builder_changed=$BUILDER_CHANGED"
```

Little suggestion to reduce repetition here:
```suggestion
          echo "dependency_changed=$DEPENDENCY_CHANGED" | tee $GITHUB_OUTPUT
          echo "builder_changed=$BUILDER_CHANGED" | tee $GITHUB_OUTPUT
```